### ISSUE TITLE
Construct Preliminary CNN Model

### ISSUE THREAD

**Issue by avzero07** [avzero07/dl-based-sensor-calib#9]
Write a basic CNN model for regression and accompanying tests.

**Comment by avzero07** [comment]
It trains!

<pre>test/network_tests/test_networks.py::test_train Start Training
Train Epoch: 1  Batch: 0        Loss: 0.008204
Train Epoch: 1  Batch: 10       Loss: 0.004348
Train Epoch: 1  Batch: 20       Loss: 0.002109
Train Epoch: 1  Batch: 30       Loss: 0.001244
Train Epoch: 1  Batch: 40       Loss: 0.001103
Train Epoch: 1  Batch: 50       Loss: 0.000880
Train Epoch: 1  Batch: 60       Loss: 0.000466
Train Epoch: 1  Batch: 70       Loss: 0.000832
Train Epoch: 1  Batch: 80       Loss: 0.001245
Train Epoch: 1  Batch: 90       Loss: 0.001826
Train Epoch: 1  Batch: 100      Loss: 0.002428
Train Epoch: 1  Batch: 110      Loss: 0.000799
Train Epoch: 1  Batch: 120      Loss: 0.000879
Train Epoch: 1  Batch: 130      Loss: 0.000690
Train Epoch: 1  Batch: 140      Loss: 0.000536
Train Epoch: 1  Batch: 150      Loss: 0.000730
Train Epoch: 1  Batch: 160      Loss: 0.000598
PASSED</pre>

**Comment by avzero07** [comment]
Still tweaking batch sizes. 30 seems to be a good number.

<pre>test/network_tests/test_networks.py::test_train Start Training
Train Epoch: 1  Batch: 0        Loss: 0.004569
Train Epoch: 1  Batch: 10       Loss: 0.003208
Train Epoch: 1  Batch: 20       Loss: 0.002262
Train Epoch: 1  Batch: 30       Loss: 0.001310
Train Epoch: 1  Batch: 40       Loss: 0.000953
Train Epoch: 1  Batch: 50       Loss: 0.000730
PASSED</pre>

**Comment by avzero07** [comment]
Preliminary model runs well. After dry run, officially at hyperparameter tuning stage!

<pre>rootdir: C:\Users\aksha\Desktop\dl-based-sensor-calib
collected 11 items

test/network_tests/test_networks.py::test_get_conv_output_size[ip0-op0] PASSED
test/network_tests/test_networks.py::test_get_conv_output_size[ip1-op1] PASSED
test/network_tests/test_networks.py::test_get_conv_output_size[ip2-op2] PASSED
test/network_tests/test_networks.py::test_get_conv_output_size[ip3-None] PASSED
test/network_tests/test_networks.py::test_get_layer_param[ip0-op0] PASSED
test/network_tests/test_networks.py::test_get_layer_param[ip1-op1] PASSED
test/network_tests/test_networks.py::test_get_layer_param[ip2-op2] PASSED
test/network_tests/test_networks.py::test_network_init PASSED
test/network_tests/test_networks.py::test_network_forward_pass Predicted = tensor([[0.1045, 0.0702]], device='cuda:0', dtype=torch.float64,
       grad_fn=<AddmmBackward>)
Actual = tensor([0.0214, 0.0216], dtype=torch.float64)
PASSED
test/network_tests/test_networks.py::test_train Start Training
Train Epoch: 1  Batch: 0        Loss: 0.002561
PASSED
test/network_tests/test_networks.py::test_inference Start Inference
Test Set Average Loss 0.0045
PASSED</pre>
